### ISSUE TITLE
indent error with ternary and brackets

### ISSUE THREAD

**Issue by piotr-s-brainhub** [eslint/eslint#9622]
There is a similar issue https://github.com/eslint/eslint/issues/8977 .
But I consider this one as another case and even if https://github.com/eslint/eslint/issues/8977 works as intended, IMO this issue is a bug.

**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 8.6.0
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "airbnb-base"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const x = Math.random() > 0.5
      ? Object.assign(
          {},
          {
            foo: 'bar',
          },
        )
      : null;

process.stdout.write(JSON.stringify(x));
```

```bash
./node_modules/.bin/eslint foo.js
```

**What did you expect to happen?**
I expected lint to pass.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  4:11  error  Expected indentation of 8 spaces but found 10   indent
  5:13  error  Expected indentation of 10 spaces but found 12  indent
  6:11  error  Expected indentation of 8 spaces but found 10   indent

✖ 3 problems (3 errors, 0 warnings)
```

When I change the indentation to:
```
const x = Math.random() > 0.5
      ? Object.assign(
        {},
        {
          foo: 'bar',
        },
      )
    : null;

process.stdout.write(JSON.stringify(x));
```
lint passes.

**Comment by gyandeeps** [comment]
Can you please share the exact config for indent rule?

**Comment by piotr-s-brainhub** [comment]
@gyandeeps 
it can be default
```
{
  "extends": "airbnb-base",
  "rules": {
    "indent": "error"
  }
}
```

**Comment by platinumazure** [comment]
@piotr-s-brainhub That configuration extends from airbnb-base so it's not clear if Airbnb is setting a different indent configuration. Note that specifying a rule with just a severity does retain the configuration from the inherited config.

Please try running `eslint --print-config path/to/file.js` and show the `indent` configuration that shows up there. Thanks!

**Comment by piotr-s-brainhub** [comment]
so it's:
```
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1,
        "VariableDeclarator": 1,
        "outerIIFEBody": 1,
        "FunctionDeclaration": {
          "parameters": 1,
          "body": 1
        },
        "FunctionExpression": {
          "parameters": 1,
          "body": 1
        }
      }
    ],
```

**Comment by piotr-s-brainhub** [comment]
even if it's changed to `"indent": ["error", 2, {}]`, eslint outputs the same errors

**Comment by not-an-aardvark** [comment]
This is working as intended. Also see: https://github.com/eslint/eslint/issues/8775
